src/transform/plugins/term/termDefinitions.ts

```diff
@@ -121,31 +130,53 @@ function processTermDefinition(
         state.env.terms[':' + label] = title;
     }
 
-    const termNodes = [];
-
+    const map: [number, number] = [startLine, endLine];
     token = new state.Token('template_open', 'template', 1);
+    token.map = map;
     token.attrSet('id', ':' + label + '_template');
-    termNodes.push(token);
+    token.attrSet('label', label);
 
-    token = new state.Token('term_open', 'dfn', 1);
+    state.tokens.push(token);
+
+    token = new state.Token('dfn_open', 'dfn', 1);
     token.attrSet('class', 'yfm yfm-term_dfn');
     token.attrSet('id', ':' + label + '_element');
     token.attrSet('role', 'tooltip');
-    termNodes.push(token);
 
-    termNodes.push(...md.parse(title, {}));
+    state.tokens.push(token);
 
-    token = new state.Token('term_close', 'dfn', -1);
-    termNodes.push(token);
+    const withOffset = md.parse(title, {}).map((el) => {
+        /**
+         * without this it doesn't work
+         */
+        if (el.children?.length) {
+            el.content = '';
+        }
 
-    token = new state.Token('template_close', 'template', -1);
-    termNodes.push(token);
+        const [start, end] = el.map || [null, null];
 
-    if (!state.env.termTokens) {
-        state.env.termTokens = [];
-    }
+        if (start === null || end === null) {
+            return el;
+        }
+
+        /**
+         * because we parse the content of terms regardless of everything,
```

Comment:
as i said i think giving more meaningful name to the function will be better than leaving such comments.